### PR TITLE
[Account] Use IArgumentCompleter to implement the Environment parameter auto completer.

### PR DESCRIPTION
**Pull request by LucasYao93**
## Description
Changed ScriptBlock to use IArgumentCompleter to implement The Environment parameter auto completer.

#15991 

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs [hunk 1]
```diff
@@ -28,13 +31,16 @@ public static string[] GetEnvironments()
...
+        /// Implementations CompleteArgument function of the <see cref="IArgumentCompleter"/>.
+        /// </summary>
+        public IEnumerable<CompletionResult> CompleteArgument(string commandName, string parameterName, string wordToComplete, CommandAst commandAst, IDictionary fakeBoundParameters)
         {
-            string script = "param($commandName, $parameterName, $wordToComplete, $commandAst, $fakeBoundParameter)\n" +
-                "$environments = [Microsoft.Azure.Commands.Profile.Common.EnvironmentCompleterAttribute]::GetEnvironments()\n" +
-                "$environments | Where-Object { $_ -Like \"$wordToComplete*\" } | ForEach-Object { [System.Management.Automation.CompletionResult]::new($_, $_, 'ParameterValue', $_) }";
-            ScriptBlock scriptBlock = ScriptBlock.Create(script);
-            return scriptBlock;
+            IEnumerable<string> names = GetEnvironments().Where(env => env.ToLower().StartsWith(wordToComplete.ToLower())) ;
```
Comment (isra-fel) [inline src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs:39]: Have you tested when you doesn't type anything and hit tab, is `wordToComplete` null or empty string? Does the completer work?

**Review by isra-fel — COMMENTED** [review]
Hey @LucasYao93 , why was the old implementation not working?

**Comment by LucasYao93** [comment]
> Hey @LucasYao93 , why was the old implementation not working?
Old implementation is work. But I think that we should to use IArgumentCompleter to implement it instead of using ScriptBlock. Because `[Microsoft.Azure.Commands.Profile.Common.EnvironmentCompleterAttribute]::GetEnvironments()`  in the ScriptBlock  maybe cause  issue when modify EnvironmentCompleterAttribute.

File: src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs (hunk 1 above)
Comment (LucasYao93) [reply src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs:39]: Tested. Can list word when doesn't type anything or hit tab.

File: src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs [hunk 2]
```diff
@@ -28,13 +31,16 @@ public static string[] GetEnvironments()
             return profileClient.ListEnvironments(null).Select(x => x.Name).ToArray();
         }
 
-        private static ScriptBlock CreateScriptBlock()
+        /// <summary>
+        /// Implementations CompleteArgument function of the <see cref="IArgumentCompleter"/>.
+        /// </summary>
+        public IEnumerable<CompletionResult> CompleteArgument(string commandName, string parameterName, string wordToComplete, CommandAst commandAst, IDictionary fakeBoundParameters)
```
Comment (isra-fel) [inline src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs:37]: As we talked offline, the test cases  should be updated to test the equality of the results of `GetEnvironments()` and `CompleteArgument()`.

Comment (LucasYao93) [reply src/Accounts/Accounts/Common/EnvironmentCompleterAttribute.cs:37]: Done

**Review by wyunchi-ms — DISMISSED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**